core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -704,6 +704,7 @@ void retry(Exception failure) {
                 return;
             }
             setPhase(task, "waiting_for_retry");
+            request.onRetry();
```

Comment:
can we add checks in the unit tests of this class (TransportReplicationActionTests) to make sure this is called?